### DISCUSSION TITLE
Restructure of Downloads

### DISCUSSION THREAD

**Discussion by toby63** [mumble-voip/mumble-www#119] [Ideas]
This will probably not be liked by everyone, but still I would like to suggest some things:

- Use OS categories (Windows, Mac, Linux, Android, iOS), Reasons:
     - Especially for linux the site now can be confusing
     - we could add some OS specific information to it (Server included in windows binary; macOS workaround if not signed correctly etc.) 
- Use a Table of Content (for better overview)
- Suggest the OS categories, at least for Linux
- Remove the section for "development snapshots" for now (can be commented) and reimplement once it's ready again
- Rewrite the "Source Code" Section:
     - implement links to the building instructions
- Add "development snapshots" to each OS category, once it's ready again.

**Comment by Krzmbrzl** [comment]
With "OS categories", do you mean individual pages or individual sections within a single page?

**Reply by toby63** [reply]
Individual sections should be enough.

But I should mention that Kissaki had a different oppinion on this, though he mentioned potential further changes in the future:
https://github.com/mumble-voip/mumble-www/pull/92

**Reply by Krzmbrzl** [reply]
Okay. I guess the overall idea here is to basically bring the notes for a specific OS together with the download button for that OS, so that the notes are readily seen when downloading Mumble?

I definitely see that this has some value. I also see why the current grid-like layout is great.

Maybe we could create a hybrid of both worlds: Keep the grid-like structure, but the links of these boxes may either directly start the download, if there are no notes for this release, or redirect to a page containing the notes and the actual download link (or redirect to the page with the notes and then start the download automatically) :thinking:

**Reply by toby63** [reply]
> I guess the overall idea here is to basically bring the notes for a specific OS together with the download button for that OS, so that the notes are readily seen when downloading Mumble?

That is one part yes.
Essentially we have a situation of two seperate perspectives:
1. Most OS's (Windows, Mac, Android, iOS) only have one download option, so for those it could work as is; nonetheless visible notes could be useful for them as well.
2. Linux: Which has multiple different methods for download, so a category for it makes the most sense.

> Keep the grid-like structure, but the links of these boxes may either directly start the download, if there are no notes for this release, or redirect to a page containing the notes and the actual download link (or redirect to the page with the notes and then start the download automatically) thinking

Those are interesting ideas.

I would use one of the following methods:
1. Put the linux grids into the Linux category.
2. Make categories for all OS's and put the grids into them (that was also my proposal last time); in addition we could have a Table of Content above.
3. Use Number 2, but instead of a classical TOC, there could be a new grid, but with only one symbole per OS, like:
    ` Windows, Linux, MacOS, iOS, Android`   and these would link to the categories below.

**Reply by Krzmbrzl** [reply]
I think I like option 3 :+1:

**Comment by toby63** [comment]
@Kissaki

**Reply by Kissaki** [reply]
* We have categories.
* ToC makes sense as it is. Maybe it can be designed better to not need it or present it differently.
* Suggestions exist #114, if that is not what you mean please provide context and be more specific
* AFAIK snapshots are working just fine
* Linking build instructions from the source code section seems like a very good idea.

**Reply by toby63** [reply]
1. Categories: What I mean is to remove this mix and put things strictly into OS categories.
3. Suggestions: What I mean is to suggest categories instead of specific binaries (especially for linux this would be better than suggesting ubuntu ppas for every linux user).
4. Snapshots: Yes now they are working again, I wrote this when they weren't.
